### PR TITLE
Advanced Installation - specify port requirements for External Load Balancer Integrations

### PR DESCRIPTION
**Pull request by mburke5678**
https://bugzilla.redhat.com/show_bug.cgi?id=1489073

### CODE REVIEW

**Comment by mburke5678** [comment]
@sjwalter2 PTAL

I added the "VIP must listen" statement as a bullet. And, I changed "pre-configured load balancing solution" to "external load balancing solution." Does this make sense? 

(Disregard the "This HAProxy load balancer" note that appears to have been added in this PR. It is from an earlier PR and I moved the note higher than previous, thus it looks like it was added).

**Comment by mburke5678** [comment]
From BZ:

https://bugzilla.redhat.com/show_bug.cgi?id=1489073#c2

**Comment by mburke5678** [comment]
@jianlinliu Changed the line to:
"A VIP listening on a port and proxying back to all master hosts on that port."

**Comment by mburke5678** [comment]
@sdodson Does this fall in your area? 
Can you help determine if the VIP *must* listen on 8443?

For your pre-configured load balancing solution, you must have:
   * A pre-created load balancer VIP configured for SSL passthrough.
   * VIP must listen on port 8443, and proxy back to all master hosts on port 8443"
   * A domain name for VIP registered in DNS.

**Comment by sdodson** [comment]
> VIP must listen on port 8443, and proxy back to all master hosts on port 8443"

The VIP must listen on whatever port is specified by the openshift_master_api_port and openshift_master_console_port values which default to 8443.

**Comment by mburke5678** [comment]
@openshift/team-documentation PTAL

**Comment by mburke5678** [comment]
@sdodson @sjwalter2 @jianlinliu Does this change affect OpenShift 3.4, 3.5, and/or 3.6?

File: install_config/install/advanced_install.adoc [hunk 1]
```diff
@@ -1865,13 +1865,22 @@ availability (HA) method:
...
+assumed you have pre-configured an external load balancing solution of your choice to
 balance the master API (port 8443) on all master hosts.
 |===
 
-For your pre-configured load balancing solution, you must have:
+[NOTE]
+====
+This HAProxy load balancer is intended to demonstrate the API server's HA mode 
+and is not recommended for production environments. If you are deploying to a cloud provider we recommend 
+that you deploy a cloud-native TCP-based load balancer or take other steps to provide a highly available load balancer.
```
Comment (bfallonf) [inline install_config/install/advanced_install.adoc]: "We" is second-person, and not recommended in the docs. Maybe:

"If you are deploying to a cloud provider, Red Hat recommends deploying a cloud-native TCP-based load balancer, or taking other steps to provide a highly available load balancer."

**Comment by bfallonf** [comment]
@mburke5678 One comment from me. LG otherwise.

**Review by sdodson — APPROVED** [review]

**Comment by mburke5678** [comment]
[rev_history]
|xref:../install_config/install/advanced_install.adoc#install-config-install-advanced-install[Advanced Installation]
|Clarified required port for VIP and added note that HAProxy router is not for production environments in xref:../install_config/install/advanced_install.adoc#multiple-masters-examples{Multiple Masters Example]. 
%

**Pull request merged**